### ISSUE TITLE
Movies don't show on the remote

### ISSUE THREAD

**Issue by GoogleCodeExporter** [peteaky/android-xbmcremote#199]
```
What steps will reproduce the problem?
1. Click the Watch your movies button
OR
1. Click the Watch your movies button
2. Go to actor tab

What is the expected output? What do you see instead?
My movies or actors. There nothing displayed and they are in the library 
and I have unseen movies(in case it does not display seen movies)

What version of XBMC Remote are you using?
0.6.6

Which Android device are you using (and which ROM, if custom)?
Nexus One 2.1 update 1 (stock)

Which XBMC revision are you running and on which platform?
XBMC 9.04.1 Stable (XBOX)

Please provide any additional information below.

I've only tried v0.6.3 and v0.6.6 and both of them dont work in actor and 
movies mode. Genre did worked for me in 0.6.3 but does not anymore due to 
another bug (#197). As for the current bug, it does not actually close or 
crash, its just not displaying anything.

```

Original issue reported on code.google.com by `steve.garon` on 24 Mar 2010 at 10:24

**Comment by GoogleCodeExporter** [comment]
```
If you close the app (Press Menu>Exit) and try again. Is the problem still 
there?
```

Original comment by `bolin.andreas@gmail.com` on 25 Mar 2010 at 12:14

**Comment by GoogleCodeExporter** [comment]
```
Yes. I have never been able to see the movies nor 
the actors on the remote. Even after a phone 
reboot.
```

Original comment by `steve.garon` on 25 Mar 2010 at 12:19

**Comment by GoogleCodeExporter** [comment]
```
If you could come to IRC we can better discuss this one. We probably need to 
add some 
more debug info in order to see what's going on. You can reach us on Freenode 
at #xbmc-
android.

```

Original comment by `phree...@gmail.com` on 25 Mar 2010 at 12:11
- Changed state: **Accepted**
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**, **Type-Enhancement**

**Comment by GoogleCodeExporter** [comment]
```
Ill tried earlier to log into IRC but no one was there. Altho, I installed 
0.6.7 and 
problem is still there. I do have a lead on what may be part of this bug. With 
the 
new version in the movie genre, when I click on a genre I get this:

[ 03-31 17:04:05.987   882:0x372 E/AndroidRuntime ]
java.lang.ClassCastException: 
org.xbmc.android.remote.presentation.widget.OneLabelItemView
at 
org.xbmc.android.remote.presentation.controller.MovieGenreListController$1.onIte
mClic
k(MovieGenreListController.java:62)
at android.widget.AdapterView.performItemClick(AdapterView.java:284)
at android.widget.ListView.performItemClick(ListView.java:3285)
at android.widget.AbsListView$PerformClick.run(AbsListView.java:1640)
at android.os.Handler.handleCallback(Handler.java:587)
at android.os.Handler.dispatchMessage(Handler.java:92)
at android.os.Looper.loop(Looper.java:123)
at android.app.ActivityThread.main(ActivityThread.java:4363)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:521)
at 
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:860)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)
at dalvik.system.NativeStart.main(Native Method)

Dont know if its related or not but it is a change from what I had before in 
this 
section. Movies files, and actor are still not showing up. I'll try to run the 
app 
into the SDK and capture the traffic to see if the XBOX returns something after 
the 
DB call.
```

Original comment by `steve.garon` on 31 Mar 2010 at 9:16

**Comment by GoogleCodeExporter** [comment]
```
Well I managed to install your latest trunk and ran it in eclipse with the 
android 
SDK 7. I captured the traffic and got this:

GET /xbmcCmds/xbmcHttp?
command=QueryVideoDatabase(SELECT+idMovie%2C+c00%2C+c07%2C+strPath%2C+strFileNam
e%2C+
c15%2C+c11%2C+c14%2C+c05+FROM+movieview+WHERE+movieview.idmovie+NOT+IN+%28SELECT
+idmo
vie+FROM+setlinkmovie%29+ORDER+BY+lower%28c00%29+ASC) HTTP/1.1
authorization: Basic XXXXXXXXXXXXXXXXXXXXXXXXXXXX
User-Agent: Dalvik/1.1.0 (Linux; U; Android 2.1; sdk Build/ERD79)
Host: XXX.XXX.XXX.XXX
Accept: *, */*
Connection: Keep-Alive

HTTP/1.0 200 OK
Server: GoAhead-Webs
Pragma: no-cache
Cache-control: no-cache
Content-Type: text/html

<html>
<li>Error:SQL error or missing database
Query: SELECT idMovie, c00, c07, strPath, strFileName, c15, c11, c14, c05 FROM 
movieview WHERE movieview.idmovie NOT IN (SELECT idmovie FROM setlinkmovie) 
ORDER BY 
lower(c00) ASC
</html>

BTW: the XXX is just me being paranoid .. 
```

Original comment by `steve.garon` on 1 Apr 2010 at 1:36

**Comment by GoogleCodeExporter** [comment]
```
Got it to work ! 

Here is my replacement function for the VideoClient.java

    public ArrayList<Movie> getMovies(INotifiableManager manager, int sortBy, 
String sortOrder) {
        StringBuilder sb = new StringBuilder();
        sb.append("SELECT idMovie, c00, c07, strPath, strFileName, c15, c11, 
c14, c05");
        sb.append(" FROM movie, files, path WHERE movie.idFile=files.idFile 
AND path.idPath=files.idPath");
        sb.append(moviesOrderBy(sortBy, sortOrder));
        return parseMovies(mConnection.query("QueryVideoDatabase", 
sb.toString(), manager));
    }

```

Original comment by `steve.garon` on 1 Apr 2010 at 2:02

**Comment by GoogleCodeExporter** [comment]
```
Ok so I fixed all the sql queries. I still can get any actors in the actor page 
but I 
realized that actors are not loaded in my DB. Which is kind of weird but thats 
not 
related to the remote control. If you check the comment 4 that I made, if you 
change 
line 62 to the following line it will fix the problem...

Genre genre = 
(Genre)mList.getAdapter().getItem(((OneLabelItemView)view).position); 

Basically what happened the LabelItemView thingy was wrong. I've attached my 
VideoClient.java.

Enjoy!

```

Original comment by `steve.garon` on 1 Apr 2010 at 2:40

Attachments:
- [VideoClient.java](https://storage.googleapis.com/google-code-attachments/android-xbmcremote/issue-199/comment-7/VideoClient.java)

**Comment by GoogleCodeExporter** [comment]
```
Sorry mate. Read you message on IRC. Will take a look at your code ASAP.

Cheers!
```

Original comment by `phree...@gmail.com` on 17 Apr 2010 at 12:24

**Comment by GoogleCodeExporter** [comment]
```
Okay, it seems your XBMC version has a different version of views. So basically 
all movie 
queries should be rewritten so we don't select the view but join it manually? 
And what about 
the ClassCastException you posted in 4? Is that still happening?

Is there a way you could run the query you pasted above that crashed directly 
on the .db file 
using an sqlite client and post the error message? It's because the view seems 
a less critical 
element to me than doing the joins manually.

On the other hand, if your code doesn't break anything with current XBMC, 
that's fine for me 
too. Can I ask you for a more recent version of those changes you've posted? A 
diff against 
current HEAD would be great.

Cheers!
```

Original comment by `phree...@gmail.com` on 17 Apr 2010 at 12:42

**Comment by GoogleCodeExporter** [comment]
```
phreezie: Yeah version 9.04 of XBMC does not have the views all. I when to the 
XBMC 
wiki and pull out the DB schema to build the queries I've used. I believe that 
it 
would be compatible with a greater number of version of XBMC if you where to 
use 
joins. That way you would have to rely on views being there or not. If you 
really 
need it, I can pull down the latest version of the code, reapply the fix and 
send 
them back as an attachment to the ticket. (I will do that later during the 
day...) If 
you want to see the error there is not need to run them using sqllite as the 
error 
was already posted on post 5. I'm fairly certain that my changes wouldn't break 
anything since I've tested the code on XBMC 9.04.2 stable on xbox and 9.11 
stable on 
my PC. I wouldn't see why XBMC's coders would want to change the DB schema 
suddenly...

As for the class exception on post 4: When I posted that I though it was 
related but 
it turned out that it had nothing to do with that. This is a totally separate 
bug. 
What appended is that when you would click on a genre, you'd get a force close. 
This 
is because the type cast of the control was for a FiveLabelItemView but the 
required 
type cast needed was a OneLabelItemView. I dunno if that has been fixed or not 
but 
this fixes it : 

Genre genre = 
(Genre)mList.getAdapter().getItem(((OneLabelItemView)view).position); 

I can pull down the latest version of the code, fix it, and post the resulting 
file 
for that bug too. I will do that later during the day. 

```

Original comment by `steve.garon` on 17 Apr 2010 at 5:31

**Comment by GoogleCodeExporter** [comment]
```
Alright, I checked out the latest code and merged my changes. I tried to test 
it the 
best that I could but got a lot of problem with this latest code since when I 
was 
navigating back to the main page I was loosing connection and that some queries 
where 
getting an Unauthorized response from the HTTP api because the basic auth 
credentials 
where not transfered during the query. Anyway, attach to this post is the 
latest 
VideoClient file with the queries fix. This should work with all versions of 
XBMC. 
Basically, the only thing that change is that I removed all movieview reference 
and 
replaced it by : 
movie, files, path WHERE movie.idFile=files.idFile AND path.idPath=files.idPath

could also be replace by (does not really matter it your preference):
movie join files on movie.idFile=files.idFile join path on 
path.idPath=files.idPath

That will make the movie code compatible with all versions of XBMC (at least 
for 
9.04.2) which to me makes more sense.

As for the class exception: Ignore it, it is already fixed in the latest code.
```

Original comment by `steve.garon` on 18 Apr 2010 at 1:17

Attachments:
- [VideoClient.java](https://storage.googleapis.com/google-code-attachments/android-xbmcremote/issue-199/comment-11/VideoClient.java)

**Comment by GoogleCodeExporter** [comment]
```
Looks good to me. Thanks!
```

Original comment by `phree...@gmail.com` on 19 Apr 2010 at 8:38

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision r601.
```

Original comment by `phree...@gmail.com` on 19 Apr 2010 at 8:40
- Changed state: **Fixed**
